webapp/channels/src/components/sidebar/sidebar_channel/sidebar_channel_menu/sidebar_channel_menu.tsx

```diff
@@ -37,7 +37,8 @@ const SidebarChannelMenu = (props: Props) => {
     let markAsReadUnreadMenuItem: JSX.Element | null = null;
     if (props.isUnread) {
         function handleMarkAsRead() {
-            props.markChannelAsRead(props.channel.id, true);
+            // We use mark multiple to not update the active channel in the server
+            props.markMultipleChannelsAsRead({[props.channel.id]: Date.now()});
```

Comment:
Which part of marking a channel as read sets the active channel? Is it something on the client or server?

I think this would be worth pulling into its own discussion since I didn't realize that marking the channel as read also updates the active channel. I'm trying to think if there's any other places that might do this which also update the active channel unintentionally

Reply:
`channels/app/channel.go` line 3032. Any time `ViewChannel` is called, we set the active channel. At least in the web client I check every time we call that API, but I may have missed something.

Reply:
Oh, I didn't realize that `markChannelAsRead` calls the `viewChannel` API. Our naming conventions here are awful because I'm pretty sure that in other cases, marking as read vs marking as viewed means and does completely different things